tests/providers/google/cloud/operators/test_automl.py

```diff
@@ -90,6 +100,22 @@ def test_execute(self, mock_hook):
             metadata=(),
         )
 
+    @mock.patch("airflow.providers.google.cloud.operators.automl.CloudAutoMLHook")
+    def test_execute_deprecated(self, mock_hook):
+        op = AutoMLTrainModelOperator(
+            model=MODEL_DEPRECATED,
+            location=GCP_LOCATION,
+            project_id=GCP_PROJECT_ID,
+            task_id=TASK_ID,
+        )
+        with pytest.raises(AirflowProviderDeprecationWarning) as err:
```

Comment:
warnings intends to be checked by the [`pytest.warns`](https://docs.pytest.org/en/stable/how-to/capture-warnings.html#warns)

Reply:
@Taragolis I was not aware of this one. Thank you for the warning. I am on it

Reply:
@Taragolis Hi again, I checked the docs and tried to use warnings (`pytest.warns`) but the thing is that we are **raising** the `AirflowProviderDeprecationWarning` in case of specific usage of the operators. I know it is a little bit different than general behavior but all these services ([AutoML](https://cloud.google.com/automl/docs) ([Tables](https://cloud.google.com/automl-tables/docs/deprecations), [Vision](https://cloud.google.com/vision/automl/docs/deprecations), [Video Intelligence](https://cloud.google.com/video-intelligence/automl/docs/deprecations), [Natural Language](https://cloud.google.com/natural-language/automl/docs/deprecations))) already are EoL but same operators can be use for AutoML Translation which is not deprecated.

Reply:
We should not raise warning as regular exception and usual it is semantically incorrect, because this is user decision whether or not raise specific warning as exception by configure `PYTHONWARNINGS` (https://docs.python.org/3/library/warnings.html)